src/weather.cpp

```diff
@@ -71,21 +72,23 @@ void weather_effect::glare()
 
 int get_hourly_rotpoints_at_temp( int temp );
 
-time_duration get_rot_since( const time_point &start, const time_point &end, const tripoint &location )
+time_duration get_rot_since( const time_point &start, const time_point &end,
+                             const tripoint &location )
 {
-
     time_duration ret = 0;
     // if underground it ignores weather, using strait underground temperature instead
-    if ( location.z < 0 ) {
-        for( time_point i = start; i < end; i += 1_hours ) {
-            ret += std::min( 1_hours, end - i ) / 1_hours * get_hourly_rotpoints_at_temp( g->get_temperature( location ) ) * 1_turns;
-        }
+    // root cellars are considered as underground storage, ignores weather, constant temperature
+    if( g->m.ter( location ) == t_rootcellar || location.z < 0 ) {
+        ret = ( end - start ) / 1_hours * get_hourly_rotpoints_at_temp( AVERAGE_ANNUAL_TEMPERATURE ) *
+              1_turns;
```

Comment:
Don't assign `ret` if you return it on the next line. Move the `ret` declaration below the block, in this block return the value you'd assign to `ret` without assigning it.

Also, check for `location.z` before checking for terrain type. If one check in `if` is orders of magnitude faster and more common than the other, it should be moved to front so that it is eliminated faster.